File: qualifier-week-5/graded-assignments/GrPA2 - Graded.go
package main

import "fmt"

func main() {
	var a, b int

	_, err := fmt.Scan(&a)
	if err != nil {
		panic(err)
	}

	_, err = fmt.Scan(&b)
	if err != nil {
		panic(err)
	}
	println(GCD(a, b))
}

func GCD(a, b int) int {
	for b != 0 {
		temp := b
		b = a % b
		a = temp
	}
	return a
}

File: qualifier-week-5/graded-assignments/GrPA2 - Graded.lua
function GCD(a, b)
    while b ~= 0 do
        local temp = b
        b = a % b
        a = temp
    end
    return a
end

local a = io.read("*n")
local b = io.read("*n")

print(GCD(a, b))

-- Recussion
--[[
function gcd(a, b)
    if b == 0 then
        return a
    else
        gcd(b, a % b)
    end
end
]]--
